app/main.py

```diff
@@ -1,2 +1,121 @@
+from typing import Any
+
+
+class HashTable:
+    def __init__(self, key: Any, value: any) -> None:
+        self.key = key
+        self.value = value
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.table = [None] * self.capacity
+        self.load_factor = 0.75
+
+    def __str__(self) -> str:
+        return f"{self.table}"
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if not hash(key):
+            raise TypeError("Dictionary key must be immutable")
+
+        index = self.get_index(key)
+
+        while self.table[index] is not None:
+            if self.table[index].key == key:
+                self.table[index].value = value
+                return
+            index = (index + 1) % self.capacity
```

Comment:
this basically is almost the same logic as in your `find_index` method. Can you refactor to follow DRY principle?